go-controller/pkg/util/ovs.go

```diff
@@ -101,6 +101,22 @@ func init() {
 		klog.Warningf("Unable to detect OS MAX_ARGS, defaulting to: %d", defaultOSMaxArgs)
 		maxArgs = defaultOSMaxArgs
 	}
+	maxArgs = findMaxArgsUsable(maxArgs)
+	klog.Infof("maximum command line arguments set to: %d", maxArgs)
+}
+
+// findMaxArgsUsable finds the maximum amount of usable args on the system, which may be
+// different than what the kernel returns for ARG_MAX
+func findMaxArgsUsable(estimatedMax int) int {
+	backoff := .9
+	args := make([]string, estimatedMax)
+	for i := range args {
+		args[i] = "a"
+	}
+	if _, err := exec.Command("/bin/true", args...).Output(); err != nil {
+		return findMaxArgsUsable(int(float64(estimatedMax) * backoff))
+	}
```

Comment:
@dcbw maybe I should add a check to ensure the estimatedMax is not < defaultOSMaxArgs (or some value) in case for some reason the bash cmd continuously errors for a different reason?